examples/src/main/python/mllib/streaming_k_means_example.py

```diff
@@ -0,0 +1,53 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+from pyspark import SparkContext
+from pyspark.streaming import StreamingContext
+# $example on$
+from pyspark.mllib.linalg import Vectors
+from pyspark.mllib.regression import LabeledPoint
+from pyspark.mllib.clustering import StreamingKMeans
+# $example off$
+
+if __name__ == "__main__":
+    sc = SparkContext(appName="StreamingKMeansExample")  # SparkContext
+    ssc = StreamingContext(sc, 1)
+
+    # $example on$
+    # we make an input stream of vectors for training,
+    # as well as a stream of labeled data points for testing
+    def parse(lp):
+        label = float(lp[lp.find('(') + 1: lp.find(')')])
+        vec = Vectors.dense(lp[lp.find('[') + 1: lp.find(']')].split(','))
+        return LabeledPoint(label, vec)
+
+    trainingData = ssc.textFileStream("data/mllib/streaming_kmeans_data.txt").map(Vectors.parse)
+    testingData = ssc.textFileStream("data/mllib/streaming_kmeans_data_test.txt").map(parse)
+
+    # We create a model with random clusters and specify the number of clusters to find
+    model = StreamingKMeans(k=2, decayFactor=1.0).setRandomCenters(3, 1.0, 0)
+
+    # Now register the streams for training and testing and start the job,
+    # printing the predicted cluster assignments on new data points as they arrive.
+    model.trainOn(trainingData)
+    print(model.predictOnValues(testingData.map(lambda lp: (lp.label, lp.features))))
```

Comment:
The `textFileStream` only reads new files in a directory. See https://github.com/apache/spark/blob/master/streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala#L79.

Maybe we can change the example with queueStream, refer to [here](https://github.com/yinxusen/spark/blob/SPARK-12042/examples/src/main/python/mllib/streaming_test_example.py).

Also, we need to change the `ssc.awaitTermination()` in the end to `ssc.stop(stopSparkContext=True, stopGraceFully=True)` ([link here](https://github.com/yinxusen/spark/blob/SPARK-12042/examples/src/main/python/mllib/streaming_test_example.py#L53)) because the former requires stop manually.